opencog/nlp/lg-parse/LGParseLink.cc

```diff
@@ -136,17 +136,14 @@ LGParseMinimal::LGParseMinimal(const Link& l)
 
 // =================================================================
 
-Handle LGParseLink::execute(AtomSpace* as) const
+Handle LGParseLink::execute() const
 {
 	if (PHRASE_NODE != _outgoing[0]->get_type()) return Handle();
 	if (LG_DICT_NODE != _outgoing[1]->get_type()) return Handle();
 	if (3 == _outgoing.size() and
 	   NUMBER_NODE != _outgoing[2]->get_type()) return Handle();
 
-	if (nullptr == as) as = getAtomSpace();
-	if (nullptr == as)
-		throw InvalidParamException(TRACE_INFO,
-			"LgParseLink requires an atomspace to parse");
+	AtomSpace* as = getAtomSpace();
```

Comment:
Actually it doesn't seem to get the atomspace here, I guess it's fine to get the atomspace from one of the  `_outgoing` atoms though that requires them to be in the atomspace in the first place, or should it be handled in the `Instantiator` instead?

Reply:
Yeah. Getting rid of the atomspace in the execute() method seemed like a good idea, based on other parts of the code. Not so much, here, where its central.

I'm looking now.